File: Backend/src/main/java/com/luckyseven/backend/sharedkernel/exception/CustomLogicException.java
package com.luckyseven.backend.sharedkernel.exception;


import lombok.Getter;
import org.springframework.http.HttpStatus;

@Getter
public class CustomLogicException extends RuntimeException {
  private final ExceptionCode exceptionCode;
  private final String detail;
  public CustomLogicException(ExceptionCode exceptionCode, String detail) {
    super(exceptionCode.getMessage());
    this.exceptionCode = exceptionCode;
    this.detail = detail;
  }

  public CustomLogicException(ExceptionCode exceptionCode) {
    super(exceptionCode.getMessage());
    this.exceptionCode = exceptionCode;
    this.detail = "";
  }

  public CustomLogicException(ExceptionCode exceptionCode, Object... args) {
    super(String.format(exceptionCode.getMessage(), args));
    this.exceptionCode = exceptionCode;
    this.detail = "";
  }

}

File: Backend/src/main/kotlin/com/luckyseven/backend/sharedkernel/exception/CustomLogicException.kt
package com.luckyseven.backend.sharedkernel.exception

class CustomLogicException : RuntimeException {
    val exceptionCode: ExceptionCode
    val detail: String?
    
    constructor(exceptionCode: ExceptionCode) : super(exceptionCode.message) {
        this.exceptionCode = exceptionCode
        this.detail = null
    }
    
    constructor(exceptionCode: ExceptionCode, detail: Any?) : super(exceptionCode.message) {
        this.exceptionCode = exceptionCode
        this.detail = detail?.toString()
    }
    
    constructor(exceptionCode: ExceptionCode, vararg args: Any) : super(String.format(exceptionCode.message, *args)) {
        this.exceptionCode = exceptionCode
        this.detail = args.joinToString(", ")
    }
} 